dot/telemetry/mailer_test.go

```diff
@@ -206,13 +177,14 @@ func TestInfiniteListener(t *testing.T) {
 func listen(w http.ResponseWriter, r *http.Request) {
 	c, err := upgrader.Upgrade(w, r, nil)
 	if err != nil {
-		log.Printf("Error %v\n", err)
+		fmt.Printf("Error %v\n", err)
```

Comment:
```suggestion
		t.Logf("Error %v\n", err)
```

Shouldn't we fail the test here? Not sure where `listen` is used though.

Reply:
We could pass a signal channel to the mock server which would send a signal when encountering an error to the calling test (close the channel or send the error encountered in an error channel). The test that launched the WS server goroutine would then error the test with the error received, stop the client and stop the server (with contexts or otherwise). That may be a bit complicated to implement here, I'm not 100% familiar with the test code here.